### PR TITLE
update to go1.12.1

### PR DESCRIPTION
**Pull request by liggitt**
/kind feature

**Does this PR introduce a user-facing change?**:
```release-note
Update to use golang 1.12.1
```

pending availability of golang:1.12.1 image and build-cross image

### CODE REVIEW

**Comment by ixdy** [comment]
/retest
why did bazel fail?

**Comment by liggitt** [comment]
Bad sha, apparently

**Comment by liggitt** [comment]
Will recheck and repush

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -21,16 +21,16 @@ http_archive(
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "6776d68ebb897625dead17ae510eac3d5f6342367327875210df44dbe2aeeb19",
-    urls = mirror("https://github.com/bazelbuild/rules_go/releases/download/0.17.1/rules_go-0.17.1.tar.gz"),
+    sha256 = "8a0a7fba4e81a7e4a0ae707f3eeeeb05c92e100b4dffbd8b9fd410e1789f1c0a",
```
Comment (dims) [inline build/root/WORKSPACE]:
```
[dims@dims-mac 21:41] ~/Downloads ⟩ sha256sum rules_go-0.17.2.tar.gz
6433336b4c5feb54e2f45df4c1c84ea4385b2dc0b6f274ec2cd5d745045eae1f  rules_go-0.17.2.tar.gz
```

Comment (liggitt) [reply build/root/WORKSPACE]: oops... for anyone wondering, `8a0a7fba4e81a7e4a0ae707f3eeeeb05c92e100b4dffbd8b9fd410e1789f1c0a` is the sha256 of the github download redirect html page

**Comment by liggitt** [comment]
the three failing jobs are pending golang:1.12.1 docker image availability and the build-cross image

**Comment by liggitt** [comment]
Closing in favor of https://github.com/kubernetes/kubernetes/pull/75413

**Pull request closed without merging**